packages/react/src/components/controls/MediaDeviceSelect.tsx

```diff
@@ -87,9 +89,12 @@ export function MediaDeviceSelect({
     }
   }, [onDeviceListChange, devices]);
 
+  React.useEffect(() => {
+    onActiveDeviceChange?.(activeDeviceId);
+  }, [activeDeviceId]);
+
   const handleActiveDeviceChange = async (deviceId: string) => {
-    setActiveMediaDevice(deviceId);
-    onActiveDeviceChange?.(deviceId);
+    await setActiveMediaDevice(deviceId, { exact: exactMatch });
   };
```

Comment:
Maybe consider inlining?

Reply:
it's now wrapped in a try/catch so leaving as is